script.js

```diff
@@ -12,20 +12,46 @@ var undoColor = new Array(9);
 var isGameOver = false;
 var counterAnimate = 0;
 var flag = false;
-
+var orginalColors = ["red", "green", "blue", "orange", "yellow", "pink", "purple", "grey"];
+var colors = ["red", "green", "blue", "orange", "yellow", "pink", "purple", "grey"];
+var noOfPlayers = "null";
+var noOfPlayersOption = document.getElementById("noOfPlayers");
 var canvas = document.getElementById("arena");
 var button = document.getElementById("undo");
 var sound = document.getElementById("sound");
 var gameArena = canvas.getContext("2d");
 canvas.addEventListener("click", gameLoop);
 button.addEventListener("click", undoGame);
+noOfPlayersOption.addEventListener("change", selectNoOfPlayers)
+
+// initialiseMatrix();
+// initialise();
 
-initialiseMatrix();
-initialise();
+function selectNoOfPlayers()
+{
+	noOfPlayers = noOfPlayersOption.value;
+	if(noOfPlayers != "null")
+	{
+		noOfPlayers = parseInt(noOfPlayers)
+		initialiseMatrix();
+		initialise();
+	} else {
+		resetCanvas();
+	}
+}
+
+function resetCanvas()
+{
+	gameArena.clearRect(0, 0, width, height);
+	clearInterval(gameTimer);
+	colors = orginalColors;
+	document.getElementById("undo").style.visibility = "hidden";
+}
 
 function initialise()
 {
-	document.getElementById("undo").style.visibility = "visible";
+
```

Comment:
why add an empty line?

Reply:
@ketanhwr  I think that was just by mistake, does the PR look good to merge?

Reply:
I've not tested it so far, can you deploy it somewhere so that I can test it? You can probably push this code to your `gh-pages` branch and give me the link.

Reply:
@ketanhwr  I have pushed the changes into my repo and hosted here https://reddysridhar53.github.io/chain-reaction
Please have a look